llvm/lib/Target/STM8/STM8RegisterInfo.cpp

```diff
@@ -27,4 +27,49 @@
 
 namespace llvm {
 
+/// Initialize register info
+/// as there's no return address register, we Initialize
+/// that with 0
+STM8RegisterInfo::STM8RegisterInfo() : STM8GenRegisterInfo(0) {}
+
+const MCPhysReg *
+STM8RegisterInfo::getCalleeSavedRegs(const MachineFunction *MF) const {
+  // there are no callee saved regs, everything is caller saved
+  static const MCPhysReg EmptySavedRegs[] = {0};
+  return EmptySavedRegs;
+}
+
+BitVector STM8RegisterInfo::getReservedRegs(const MachineFunction &MF) const {
+  BitVector Reserved(getNumRegs());
+
+  Reserved.set(STM8::R_PC);
+  Reserved.set(STM8::R_PCE);
+  Reserved.set(STM8::R_PCH);
+  Reserved.set(STM8::R_PCL);
+  Reserved.set(STM8::R_PCHL);
```

Comment:
I just noticed that AVR doesn't mark PC as reserved (and in fact, it doesn't even define it at all in AVRRegisterInfo.td!). This is probably because on AVR there's no way to directly interact with PC except through branch instructions. You'll never see it as an instruction operand. This is also true for STM8 so we might be able to get away with not defining PC at all.

Reply:
Can you give me an example for such an instruction in e.g. ARM or X86?